core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -191,4 +191,384 @@ public UnknownException(Throwable cause) {
+         */
+        public static class FailedAttemptsLimitExceededException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "User has made too many failed attempts for a given action.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public FailedAttemptsLimitExceededException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because user made too many requests for a given operation.
+         */
+        public static class RequestLimitExceededException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "The user has made too many requests for a given operation.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public RequestLimitExceededException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because Amazon Cognito service encounters an invalid AWS Lambda response
+         * or encounters an unexpected exception with the AWS Lambda service.
+         */
+        public static class LambdaException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE =
+                    "Amazon Cognito service encounters an invalid AWS Lambda response " +
+                            "or encounters an unexpected exception with the AWS Lambda service.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public LambdaException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because device is not tracked.
+         */
+        public static class DeviceNotTrackedException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Device is not tracked.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
```

Comment:
Not a good recovery suggestion - not sure how this exception would get called but should give a recovery suggestion that would help them avoid getting this.

Reply:
Will change to "Please check out the device configuration."